src/libs_3rdparty/QSpec/src/primitives/GTWidget.h

```diff
@@ -148,6 +150,35 @@ class HI_EXPORT GTWidget {
     }
 #undef GT_METHOD_NAME
 
+#define GT_METHOD_NAME "findChildren"
+    /**
+     * Finds all children of the 'parent' using 'findChildren' method and checkFn to check if the child is matched.
+     * If parent is null, find all child in all main windows.
+     * Note: this function does not wait until any child is found and returns immediately.
+     */
+    template<class ChildType>
+    static QList<ChildType *> findChildren(GUITestOpStatus &os, const QObject *parent, std::function<bool(ChildType *)> checkFn) {
+        QList<ChildType *> children;
+        auto checkAndAddToList = [&children, &checkFn](const QList<ChildType *> candidates) -> void {
```

Comment:
nit: not need to text `void` as a return value - the compiler figure it out

Reply:
I do not want a compiler to use some implicit type it can derive: I want the return to be explicit and and make the compiler fail anytime I return a different value.

So if I add a line into the method like "return 0" I want this issue to be cached by the compiler. This way I do always add types to lambdas: "type system" is a good tool and it should not be ignored.

It may be easier to understand if you compare this 'tool' to 'override' keyword. We can also skip it but for some good reason we don't

Reply:
I read more about lambdas & code style recommendations for them and I want to say that now I agree, that is is OK to skip explicit types for a single statement lambdas. I will update my patch before submission.